### PR TITLE
Refactor odometry in SPG optimization problem

### PR DESCRIPTION
**Pull request by ojura**
[Rendered](https://github.com/larics/rfcs/blob/odometry-refactoring/text/0000-odometry-refactoring.md)

### CODE REVIEW

File: text/0000-odometry-refactoring.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+## Summary
+[summary]: #summary
+
+Do not use `std::vector<TransformInterpolationBuffer>` in SPG optimization problem anymore.
+
+## Motivation
+[motivation]: #motivation
+
+`TransformInterpolationBuffer` does not support trimming-in-middle.
```
Comment (wohe) [inline text/0000-odometry-refactoring.md:11]: I think this RFC is too fine grained and this is also not the true motivation: why do we want trimming in the middle in the first place? I agree that we should have an RFC for https://github.com/googlecartographer/cartographer/projects/1 which motivates it. @cschuet Opinion?

Comment (ojura) [reply text/0000-odometry-refactoring.md:11]: You mentioned yesterday that you want to switch away from `TransformInterpolationBuffer` and follow how IMU data is now kept in `MapByTime`.

Comment (wohe) [reply text/0000-odometry-refactoring.md:11]: "Motivation" should be what caused us to get interested in this change. Trimming-in-the-middle is not a goal by itself, but a necessity to support life-long mapping.

Comment (ojura) [reply text/0000-odometry-refactoring.md:11]: I see. If you want, I can close this PR since it's too fine grained for an RFC and open an issue.

Comment (wohe) [reply text/0000-odometry-refactoring.md:11]: No need for an issue. We have a project already (see above) which predates the RFC process. We could add an RFC for it. Other than that, we will just do it.

Comment (ojura) [reply text/0000-odometry-refactoring.md:11]: The Projects UI is pretty much read-only for non-maintainers. Where will discussion take place?

> Other than that, we will just do it.

You mean you will do the refactoring? :-)

Comment (wohe) [reply text/0000-odometry-refactoring.md:11]: Yes. Already on it.

**Comment by SirVer** [comment]
@ojura Can this be closed? It seems like it is about a refactoring that did not affect the public API. 
@wohe Opinions?

**Comment by wohe** [comment]
I think this should be closed. The specific change it mentioned is implemented now, and if we would like to have an RFC for life-long mapping we should have a separate PR for it.

**Comment by ojura** [comment]
For the future, should I just open issues for minor stuff like this?

**Comment by wohe** [comment]
Not in my opinion. There should be RFCs on a higher level explaining the motivation for the change. The change itself should then be done in a PR. No need to make the process heavier. @cschuet What do you think?

**Pull request closed without merging**